tests/integration/helm/upgrade/util.go

```diff
@@ -168,6 +168,69 @@ func performInPlaceUpgradeFunc(previousVersion string, isAmbient bool) func(fram
 	}
 }
 
+// upgradeAllButZtunnel returns the provided function necessary to run inside an integration test
+// for upgrading everythying but the ztunnel
+func upgradeAllButZtunnel(previousVersion string) func(framework.TestContext) {
+	return func(t framework.TestContext) {
+		isAmbient := true
+		cs := t.Clusters().Default().(*kubecluster.Cluster)
+		h := helm.New(cs.Filename())
+		nsConfig := helmtest.DefaultNamespaceConfig
+		t.CleanupConditionally(func() {
+			// only need to do call this once as helm doesn't need to remove
+			// all versions
+			helmtest.DeleteIstio(t, h, cs, nsConfig, isAmbient)
+		})
+		s := t.Settings()
+		prevVariant := s.Image.Variant
+		// Istio 1.21 ambient did not support distroless, always use debug.
+		// TODO(https://github.com/istio/istio/issues/50387) remove this, always use s.Image.Variant
+		if isAmbient {
+			prevVariant = "debug"
+		}
```

Comment:
```suggestion
```

I think this entire block can/should be removed? For at least the last 2 releases the `debug` override should definitely not be required.